OrbitCore/include/OrbitCore/Error.h

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_CORE_ERROR_H_
+#define ORBIT_CORE_ERROR_H_
+
+#include <system_error>
+
+namespace OrbitCore {
```

Comment:
I need some context on this - why is this better than just having arbitrary string message in the error?

Reply:
The idea behind std::error_code is to keep the overhead of error-handling low. Having a small type which basically stores only an integer (the actual error code) and a pointer to a error category (which holds more information about the error) allows the return type to be stored in two registers instead of on the stack. It especially minimizes overhead when returning `outcome::result<void>`, where the non-error-path does not need any storage.

Another advantage of std::error_code: You can extend it without the use of RTTI (like with Exceptions). So it also works with projects that have RTTI disabled.

Reply:
I filed a bug about a unified error class. b/160554376